labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise02.scala

```diff
@@ -3,32 +3,30 @@ package org.scalalabs.basic.lab02
 import scala.collection.mutable.ListBuffer
  import sys._
 
-
+//Modified by Sisi Cheng & Jean Sung
 object ListManipulationExercise02 {
 
   /**
    * Find the maximum element in a list, e.g. maxElementInList(List(1,9,3,5)) == 9
    * As usual, various ways exist: pattern matching, folding, ...
    */
-  def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
-  }
+  def maxElementInList(l: List[Int]): Int = l.max
 
   /**
    * Calculate the sum of the equally position elements
    * of the two list
    */
   def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = {
-    error("fix me")
+    if (l1.isEmpty) l2
+    else if (l2.isEmpty) l1
+    else List.concat(List(l1.head + l2.head), sumOfTwo(l1.tail, l2.tail))
   }
 
   /**
    *  For this exercise preferably make use of the sumOfTwo
    * method above
    */
-  def sumOfMany(l: List[Int]*): List[Int] = {
-    error("fix me")
-  }
+  def sumOfMany(l: List[Int]*): List[Int] = l.fold(List())((r,c) => sumOfTwo(r,c))
```

Comment:
Your `fold` solution is better than my `reduce` solution; I forgot that `reduce` doesn’t work on the empty list.